src/mumble/Overlay.cpp

```diff
@@ -185,11 +185,45 @@ QRectF OverlayGroup::boundingRect() const {
 }
 
 Overlay::Overlay() : QObject() {
-	d = NULL;
+	d = nullptr;
 
-	platformInit();
-	forceSettings();
+	m_initialized.store(false);
 
+	qlsServer = nullptr;
+
+	QMetaObject::connectSlotsByName(this);
+}
+
+Overlay::~Overlay() {
+	setActive(false);
+	if (d) {
+		delete d;
+	}
+
+	// Need to be deleted first, since destructor references lingering QLocalSockets
+	foreach(OverlayClient *oc, qlClients)
+	{
+		// As we're the one closing the connection, we do not need to be
+		// notified of disconnects. This is important because on disconnect we
+		// also remove (and 'delete') the overlay client.
+		disconnect(oc->qlsSocket, SIGNAL(disconnected()), this, SLOT(disconnected()));
+		disconnect(oc->qlsSocket, SIGNAL(error(QLocalSocket::LocalSocketError)), this, SLOT(error(QLocalSocket::LocalSocketError)));
+		delete oc;
+	}
+}
+
+void Overlay::setActive(bool act) {
+	if (!m_initialized.load()) {
+		platformInit();
+		forceSettings();
+	
+		m_initialized.store(true);
+	}
+
+	setActiveInternal(act);
+}
+
+void Overlay::createPipe() {
```

Comment:
*Regression:* `createPipe()` is introduced but never called.

Reply:
Good catch - (hopefully) fixed in #4281